documentation/modules/overview/con-key-features-kafka-connect.adoc

```diff
@@ -0,0 +1,141 @@
+In this case, the external data system is another Kafka cluster.
+Specialized connectors for MirrorMaker 2.0 manage data replication between source and target Kafka clusters.
+
+[NOTE]
+--
+Kafka provides two built-in example connectors:
+
+* `FileStreamSourceConnector` streams data from an external system to Kafka, reading lines from an input source and sending each line to a Kafka topic.
+* `FileStreamSinkConnector` streams data from Kafka to an external system, reading messages from a Kafka topic and creating a line for each in an output file.
+--
+
+The following source connector diagram shows the process flow for a source connector that streams records from an external data system.
+A Kafka Connect cluster might operate source and sink connectors at the same time.
+Workers are running in distributed mode in the cluster.
+Workers can run one or more tasks for more than one connector instance.
+
+.Source connector streaming data to Kafka
+image:overview/kafka-concepts-source-connector.png[Kafka Connect source connector worker interaction in distributed mode]
+
+. A plugin provides the implementation artifacts for the source connector
+. A single worker initiates the source connector instance
+. The source connector creates the tasks to stream data
+. Tasks run in parallel to poll the external data system and return records
+. Transforms adjust the records, such as filtering or relabelling them
+. Converters put the records into a format suitable for Kafka
+. The source connector is managed using KafkaConnectors or the Kafka Connect API
+
+The following sink connector diagram shows the process flow when streaming data from Kafka to an external data system.
+
+.Sink connector streaming data from Kafka
+image:overview/kafka-concepts-sink-connector.png[Kafka Connect sink connector worker interaction in distributed mode]
+
+. A plugin provides the implementation artifacts for the sink connector
+. A single worker initiates the sink connector instance
+. The sink connector creates the tasks to stream data
+. Tasks run in parallel to poll Kafka and return records
+. Converters put the records into a format suitable for the external data system
+. Transforms adjust the records, such as filtering or relabelling them
+. The sink connector is managed using KafkaConnectors or the Kafka Connect API
+
+== Tasks
+
+Data transfer orchestrated by the Kafka Connect runtime is split into tasks that run in parallel.
+A task is started using the configuration supplied by a connector instance.
+Kafka Connect distributes tasks to workers.
+
+* A source connector task polls the external data system and returns a list of records that a worker sends to the Kafka brokers.
+* A sink connector task receives Kafka records from a worker for writing to the external data system.
+
+If a task fails, it can be restarted through the Kafka Connect API.
+
+For sink connectors, the number of tasks created relates to the number of partitions being consumed.
```

Comment:
I do wonder whether using the word "partition" is really the right choice here. Connect uses the work "partition" to cover both Kafka topic partitions and also how the external system's data might be split up. This can be really confusing. It might have be better if connect had used a different word than partition (e.g. shard) when talking about external systems. At least that way partition would have been unambiguously a Kafka concept. The decision we have to make here is whether to deviate from the precedent set by Kafka, or just to call out that partition can refer to a shard of _any_ data source, Kafka or not. I guess the latter is the least worst option.

Reply:
Maybe a note:

> NOTE: In the context of Kafka Connect, a _partition_ can mean a topic partition or a _shard of data_ intended for an external system.